hypershift-operator/main.go

```diff
@@ -97,17 +107,65 @@ func NewStartCommand() *cobra.Command {
 			os.Exit(1)
 		}
 
+		// Add some flexibility to getting the control plane operator image. Use the
+		// flag if given, but if that's empty and we're running in a pod, use the
+		// hypershift operator's image for the control plane by default.
+		lookupControlPlaneOperatorImage := func(kubeClient client.Client) (string, error) {
+			if len(controlPlaneOperatorImage) > 0 {
+				return controlPlaneOperatorImage, nil
+			}
+
+			podName := os.Getenv("MY_POD_NAME")
+			podNamespace := os.Getenv("MY_POD_NAMESPACE")
+			runningInPod := len(podName) > 0 && len(podNamespace) > 0
+			if !runningInPod {
+				return "", nil
+			}
+
+			var image string
+			pod := corev1.Pod{}
+			if err := kubeClient.Get(context.TODO(), client.ObjectKey{Namespace: podNamespace, Name: podName}, &pod); err != nil {
+				return "", fmt.Errorf("failed to get operator's pod %s/%s: %w", podNamespace, podName, err)
+			}
+			for _, container := range pod.Spec.Containers {
+				if container.Name == "operator" {
+					image = container.Image
+					break
+				}
+			}
+			return image, nil
+		}
+
+		// For now read release info from a file to keep it simple and externalize
+		// the complexity of the ways we currently know to get the data (which involve
+		// container runtime interactions via `oc adm release info`, etc.)
```

Comment:
just a clarification ... there is no container runtime interaction in `oc adm release info`. It only uses the image registry API to fetch layers, untar and extract files.

Reply:
Updated comment, please fact-check